File: web/src/main/resources/js-app/application/user-data.js
/**
 * Contains valid data for authorization.
 */
export class UserData {
  #email;
  #password;

  /**
   * @param {string} email
   * @param {string} password
   */
  constructor(email, password) {
    this.#email = email;
    this.#password = password;
  }

  /**
   * @returns {string}
   */
  get email() {
    return this.#email;
  }

  /**
   * @returns {string}
   */
  get password() {
    return this.#password;
  }
}

File: persistent/src/main/java/com/teamdev/filehub/database/data/UserData.java
package com.teamdev.filehub.database.data;

import com.google.common.base.Preconditions;

import javax.annotation.ParametersAreNonnullByDefault;


/**
 * Storing  information about user
 */
public class UserData extends Data {

    private final String email;

    private final String password;

    @ParametersAreNonnullByDefault
    public UserData(String id, String email, String password) {
        super(id);
        Preconditions.checkNotNull(id);
        Preconditions.checkNotNull(email);
        Preconditions.checkNotNull(password);

        this.email = email;

        this.password = password;
    }

    public String getEmail() {
        return email;
    }


    public String getPassword() {
        return password;
    }


}
